tests/unit/utils/test_analytics_utils.py

```diff
@@ -11,16 +11,157 @@
+
+    gc = GlobalConfiguration()
+    assert user_id == str(gc.user_id)
+
+    # Check the validity of the event
+    assert event in AnalyticsEvent.__members__.values()
+
+    # Check the type of each metadata entry
+    assert all([not isinstance(v, UUID) for v in metadata.values()])
+
+    # Check whether common parameters are included in the metadata
+    assert "environment" in metadata
+    assert "python_version" in metadata
+    assert "version" in metadata
+    assert "event_success" in metadata
+
+    from zenml.client import Client
+
+    client = Client()
+
+    if event == AnalyticsEvent.REGISTERED_STACK:
+
+        assert "user_id" in metadata
+        assert metadata["user_id"] == str(client.active_user.id)
+
+        assert "project_id" in metadata
+        assert metadata["project_id"] == str(client.active_project.id)
+
+        assert "entity_id" in metadata
+
+        assert StackComponentType.ARTIFACT_STORE in metadata
+        assert StackComponentType.ORCHESTRATOR in metadata
+        assert "is_shared" in metadata
+
+    if event == AnalyticsEvent.REGISTERED_STACK_COMPONENT:
+        assert "user_id" in metadata
+        assert metadata["user_id"] == str(client.active_user.id)
+
+        assert "type" in metadata
+        assert metadata["type"] in StackComponentType.__members__.values()
+
+        assert "flavor" in metadata
+        assert "entity_id" in metadata
+        assert "is_shared" in metadata
+
+    if event == AnalyticsEvent.RUN_PIPELINE:
+        assert "store_type" in metadata
+        assert metadata["store_type"] in StoreType.__members__.values()
+
+        assert "artifact_store" in metadata
+        assert (
+            metadata["artifact_store"]
+            == client.active_stack_model.components[
+                StackComponentType.ARTIFACT_STORE
+            ][0].flavor
+        )
+
+        assert "orchestrator" in metadata
+        assert (
+            metadata["orchestrator"]
+            == client.active_stack_model.components[
+                StackComponentType.ORCHESTRATOR
+            ][0].flavor
+        )
+
+
+def event_context_exit(
+    _,
+    exc_type: Optional[Type[BaseException]],
+    exc_val: Optional[BaseException],
+    exc_tb: Optional[TracebackType],
+):
+    """Mock exit function to replace the exit method of the AnalyticsContext.
+
+    Normally, the analytics context has an exit function which allows the main
+    thread to continue by suppressing exceptions which occur during the event
+    tracking. However, for the sake of the tests we would like to disable this
+    behaviour, so we can catch the errors which happen during this process.
+    """
+
+
+@patch.dict(
+    "os.environ", {"ZENML_DEBUG": "true", "ZENML_ANALYTICS_OPT_IN": "true"}
+)
+def test_analytics_event(
+    mocker: MockFixture, clean_client, one_step_pipeline, empty_step
+) -> None:
+    """Checks whether the event sent for analytics has the right properties."""
```

Comment:
As this departs from our other tests it might make sense to describe a bit how this test works and that the assertions take place in the mocked event_context_method

Reply:
Added a comprehensive docstring. Hope it helps.